### PR TITLE
[CAS-1155] Bump sdk to 31

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
[CAS-1155](https://stream-io.atlassian.net/browse/CAS-1155)

### 🎯 Goal

Bump SDK to 31

### 🛠 Implementation details

This PR comments out Flipper from the sample app, because it causes problems in the merge of manifests. Apperently it adds activities that don't have `android:exposed` in its definition.

### 🧪 Testing

Test the app in a broad way.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- ~[ ] New code is covered by unit tests~
- ~[ ] Comparison screenshots added for visual changes~
- ~[ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))~
- [x] Reviewers added

### 🎉 GIF
giphy (2)

### CODE REVIEW

File: stream-chat-android-ui-components-sample/build.gradle [hunk 1]
```diff
@@ -161,10 +161,10 @@ dependencies {
     implementation Dependencies.coilGif
     implementation Dependencies.shimmer
 
-    debugImplementation Dependencies.flipper
-    debugImplementation Dependencies.flipperNetwork
-    debugImplementation Dependencies.flipperLoader
-    debugImplementation Dependencies.fragmentTest
+//    debugImplementation Dependencies.flipper
```
Comment (samiuelson) [inline stream-chat-android-ui-components-sample/build.gradle:164]: Why flipper stuff is commented?

File: stream-chat-android-ui-components-sample/build.gradle [hunk 2]
```diff
@@ -161,10 +161,10 @@ dependencies {
...
+//    debugImplementation Dependencies.flipperNetwork
+//    debugImplementation Dependencies.flipperLoader
+//    debugImplementation Dependencies.fragmentTest
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components-sample/build.gradle:167]: I tested and Flipper works fine. The problem is caused by `Dependencies.fragmentTest`

Comment (leandroBorgesFerreira) [reply stream-chat-android-ui-components-sample/build.gradle:167]: Oh, nice. Let me test this better then.

File: stream-chat-android-ui-components-sample/build.gradle (hunk 1 above)
Comment (leandroBorgesFerreira) [reply stream-chat-android-ui-components-sample/build.gradle]: No commented anymore.

**Review by samiuelson — APPROVED** [review]

**Review by ogkuzmin — APPROVED** [review]

File: stream-chat-android-ui-components-sample/src/debug/kotlin/io/getstream/chat/ui/sample/application/ApplicationConfigurator.kt [hunk 3]
```diff
@@ -17,15 +17,15 @@ object ApplicationConfigurator {
 
     @OptIn(InternalStreamChatApi::class)
     fun configureApp(application: Application) {
-        SoLoader.init(application, false)
+        // SoLoader.init(application, false)
```
Comment (zsmb13) [inline stream-chat-android-ui-components-sample/src/debug/kotlin/io/getstream/chat/ui/sample/application/ApplicationConfigurator.kt:20]: Do we still need to have all of these things commented out?

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,7 +3,7 @@
 ### 🐞 Fixed
 
 ### ⬆️ Improved
-
+- Bumping SDKs target api to 31
```
Comment (zsmb13) [inline CHANGELOG.md:6]:
```suggestion
- Bumped the SDKs target API to 31
```

File: buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt [hunk 5]
```diff
@@ -79,7 +79,7 @@ object Versions {
     internal const val THREETENBP = "1.5.1"
     internal const val THREETENABP = "1.3.1"
     internal const val TIMBER = "5.0.0"
-    internal const val WORK = "2.5.0"
+    internal const val WORK = "2.7.0"
```
Comment (zsmb13) [inline buildSrc/src/main/kotlin/io/getstream/chat/android/Dependencies.kt:82]: Let's add a note about this in the changelog, something like

```
- Updated WorkManager to version 2.7.0, which fixes compatibility issues with SDK 31
```

File: stream-chat-android-ui-components-sample/build.gradle [hunk 6]
```diff
@@ -155,7 +155,7 @@ dependencies {
     implementation Dependencies.navigationUIKTX
 
     implementation Dependencies.firebaseAnalytics
-    implementation Dependencies.firebaseCrashlytics
+    implementation Dependencies .firebaseCrashlytics
```
Comment (zsmb13) [inline stream-chat-android-ui-components-sample/build.gradle:158]: Why is this random line change on this PR? 😃

Comment (zsmb13) [reply stream-chat-android-ui-components-sample/build.gradle:158]:
```suggestion
    implementation Dependencies.firebaseCrashlytics
```

**Review by zsmb13 — APPROVED** [review]

**Pull request merged**